print-ip-demux.c

```diff
@@ -208,6 +208,10 @@ ip_demux_print(netdissect_options *ndo,
 		pgm_print(ndo, bp, length, iph);
 		break;
 
+	case IPPROTO_ETHERNET:
+		ether_print(ndo, bp, length, ND_BYTES_AVAILABLE_AFTER(bp), NULL, NULL, FALSE);
```

Comment:
IP version can be 4 or 6 ?

Reply:
No. IPPROTO_ETHERNET indicates that the traffic carried inside the SRv6 tunnel is L2 frame. So it doesn't matter if it is IPv4 or IPv6.

Reply:
Thus we can have:
Ether/IPv6/Ether/IPv[46]
or
Ether/IPv4/Ether/IPv[46] ?
or only 
Ether/IPv6/Ether/IPv[46]

Reply:
In other words, should we have?
```patch
diff --git a/ipproto.c b/ipproto.c
index 58ceeeba..a3b0714a 100644
--- a/ipproto.c
+++ b/ipproto.c
@@ -50,6 +50,7 @@ const struct tok ipproto_values[] = {
     { IPPROTO_PGM, "PGM" },
     { IPPROTO_SCTP, "SCTP" },
     { IPPROTO_MOBILITY, "Mobility" },
+    { IPPROTO_ETHERNET, "Ethernet" },
     { 0, NULL }
 };
 
diff --git a/print-ip-demux.c b/print-ip-demux.c
index 308529b8..8b53187c 100644
--- a/print-ip-demux.c
+++ b/print-ip-demux.c
@@ -209,7 +209,13 @@ again:
 		break;
 
 	case IPPROTO_ETHERNET:
-		ether_print(ndo, bp, length, ND_BYTES_AVAILABLE_AFTER(bp), NULL, NULL, FALSE);
+		if (ver == 6)
+			ether_print(ndo, bp, length, ND_BYTES_AVAILABLE_AFTER(bp), NULL, NULL);
+		else {
+			ND_PRINT("[%s requires IPv6]",
+				 tok2str(ipproto_values,"unknown",nh));
+			nd_print_invalid(ndo);
+		}
 		break;
 
 	case IPPROTO_NONE:
```

Reply:
We can have only this case. 
**Ether/IPv6/Ether/IPv[46]**

Reply:
> Thus we can have:
> Ether/IPv6/Ether/IPv[46]
> or
> Ether/IPv4/Ether/IPv[46] ?
> or only
> Ether/IPv6/Ether/IPv[46] ?

We can have only this case.
**Ether/IPv6/Ether/IPv[46]**

Reply:
OK, thus the above patch is needed.

Reply:
Ok. will include it in the update PR.